src/recommendpolicy/helperFunctions.go

```diff
@@ -127,25 +206,14 @@ func generateKyvernoPolicy(name, namespace string, labels LabelMap) ([]kyvernov1
 }
 
 // createRestrictAutomountSATokenPolicy modifies and converts the original policy matching on pods to be suitable for Deployment
-func createRestrictAutomountSATokenPolicy(ms types.MatchSpec, name, namespace string, labels LabelMap) kyvernov1.Policy {
-	policy := kyvernov1.Policy{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "Policy",
-			APIVersion: "kyverno.io/v1",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name: name + "-" + ms.Name,
-			Annotations: map[string]string{
-				"recommended-policies.kubearmor.io/title":                "Restrict Auto-Mount of Service Account Tokens",
-				"policies.kyverno.io/minversion":                         "1.6.0",
-				"recommended-policies.kubearmor.io/description-detailed": ms.Description.Detailed,
-				"recommended-policies.kubearmor.io/tags":                 strings.Join(ms.KyvernoPolicyTags, ","),
-				"recommended-policies.kubearmor.io/description":          ms.Description.Tldr,
-			},
-			Namespace: namespace,
-		},
-	}
-	policySpec := ms.KyvernoPolicySpec.DeepCopy()
+func createRestrictAutomountSATokenPolicy(ms types.MatchSpec, name, namespace string, labels LabelMap) kyvernov1.PolicyInterface {
+	policyInterface := *(ms.KyvernoPolicy)
+	policy := (policyInterface.(*kyvernov1.Policy)).DeepCopy()
+	policy.Annotations[RecommendedPolicyTagsAnnotation] = strings.Join(ms.KyvernoPolicyTags, ",")
```

Comment:
Breaks backward compatibility with policy-templates. This would only work with the latest version of policy templates i.e. after merger of https://github.com/kubearmor/policy-templates/pull/1039